include/boost/geometry/formulas/differential_quantities.hpp

```diff
@@ -0,0 +1,269 @@
+            //CT const sin_alp2 = sin(reverse_azimuth);
+            CT const cos_alp2 = cos(reverse_azimuth);
+
+            CT sin_sig1 = sin_bet1;
+            CT cos_sig1 = cos_alp1 * cos_bet1;
+            CT sin_sig2 = sin_bet2;
+            CT cos_sig2 = cos_alp2 * cos_bet2;
+
+            normalize(sin_sig1, cos_sig1);
+            normalize(sin_sig2, cos_sig2);
+
+            CT const sin_alp0 = sin_alp1 * cos_bet1;
+            CT const cos_alp0_sqr = 1 - math::sqr(sin_alp0);
+
+            CT const J12 = BOOST_GEOMETRY_CONDITION(ApproxF) ?
+                           J12_f(sin_sig1, cos_sig1, sin_sig2, cos_sig2, cos_alp0_sqr, f) :
+                           J12_ep_sqr(sin_sig1, cos_sig1, sin_sig2, cos_sig2, cos_alp0_sqr, ep2) ;
+
+            CT const dn1 = math::sqrt(1 + e2 * math::sqr(sin_lat1));
+            CT const dn2 = math::sqrt(1 + e2 * math::sqr(sin_lat2));
+
+            if (BOOST_GEOMETRY_CONDITION(EnableReducedLength))
+            {
+                CT const m12_b = dn2 * (cos_sig1 * sin_sig2)
+                               - dn1 * (sin_sig1 * cos_sig2)
+                               - cos_sig1 * cos_sig2 * J12;
+                CT const m12 = m12_b * b;
+
+                reduced_length = m12;
+            }
+
+            if (BOOST_GEOMETRY_CONDITION(EnableGeodesicScale))
+            {
+                CT const cos_sig12 = cos_sig1 * cos_sig2 + sin_sig1 * sin_sig2;
+                CT const t = ep2 * (cos_bet1 - cos_bet2) * (cos_bet1 + cos_bet2) / (dn1 + dn2);
+                CT const M12 = cos_sig12 + (t * sin_sig2 - cos_sig2 * J12) * sin_sig1 / dn1;
+
+                geodesic_scale = M12;
+            }
+        }
+    }
+
+private:
+    /*! Approximation of J12, expanded into taylor series in f
+        Maxima script:
+        ep2: f * (2-f) / ((1-f)^2);
+        k2: ca02 * ep2;
+        assume(f < 1);
+        assume(sig > 0);
+        I1(sig):= integrate(sqrt(1 + k2 * sin(s)^2), s, 0, sig);
+        I2(sig):= integrate(1/sqrt(1 + k2 * sin(s)^2), s, 0, sig);
+        J(sig):= I1(sig) - I2(sig);
+        S: taylor(J(sig), f, 0, 3);
+        S1: factor( 2*integrate(sin(s)^2,s,0,sig)*ca02*f );
+        S2: factor( ((integrate(-6*ca02^2*sin(s)^4+6*ca02*sin(s)^2,s,0,sig)+integrate(-2*ca02^2*sin(s)^4+6*ca02*sin(s)^2,s,0,sig))*f^2)/4 );
+        S3: factor( ((integrate(30*ca02^3*sin(s)^6-54*ca02^2*sin(s)^4+24*ca02*sin(s)^2,s,0,sig)+integrate(6*ca02^3*sin(s)^6-18*ca02^2*sin(s)^4+24*ca02*sin(s)^2,s,0,sig))*f^3)/12 );
+    */
+    static inline CT J12_f(CT const& sin_sig1, CT const& cos_sig1,
+                           CT const& sin_sig2, CT const& cos_sig2,
+                           CT const& cos_alp0_sqr, CT const& f)
+    {
+        if (Order == 0)
+        {
+            return 0;
+        }
+
+        CT const sig_12 = atan2(cos_sig1 * sin_sig2 - sin_sig1 * cos_sig2,
+                                cos_sig1 * cos_sig2 + sin_sig1 * sin_sig2);
+        CT const sin_2sig1 = 2 * cos_sig1 * sin_sig1; // sin(2sig1)
```

Comment:
Should be CT(2) instead of 2. This should hold in general for all constants in the code in order to support other number types like ttmath.